dynamo/config/filters/bilateral.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
```

Comment:
I noticed that it is consuming a lot of memory. With batch size = 9, resolution = 512,512, kernel size = 25 this use 32Gb + 32Gb on swap running on CPU -- and for the GPU also consumed my 12Gb of VRAM (with smaller batch and smaller resolution) cc\ @gau-nernst

i can't replicate the results of https://github.com/kornia/kornia/issues/702#issuecomment-1451323769 using our runner script

As a side note, I expected, bilateral on OpenCV should perform better with small kernels (3 and 5) -- [it's the recommended kernel for real-time applications](https://docs.opencv.org/4.x/d4/d86/group__imgproc__filter.html#ga9d7064d478c95d60003cf839430737ed).

The arguments of the graph is in the order: bs, res, kernel size, sigma color, sigma space

![bilateral](https://user-images.githubusercontent.com/20444345/224548225-a91356d0-75c1-4170-8a7a-4538776bf561.png)

Reply:
Because the implementation operates on unfolded tensor (B, C, H, W, Kx, Ky), excessive memory consumption is expected. I think torch dynamo should be able to optimize not to allocate too much temporary memory.

I'm quite surprised that your script does not see an improvement with dynamo. Also, my previous benchmark results might not be too reliable since I might have made some mistakes. I added updated benchmark results https://github.com/kornia/kornia/issues/702#issuecomment-1451323769. If I scale the numbers correctly, I think your CPU results are faster than mine, and my GPU results are faster than yours. The difference could be due to some mistakes in my timing script (you can check in the comment) or different hardware.

Reply:
>Because the implementation operates on unfolded tensor (B, C, H, W, Kx, Ky), excessive memory consumption is expected. I think torch dynamo should be able to optimize not to allocate too much temporary memory.

I see


> I'm quite surprised that your script does not see an improvement with dynamo. Also, my previous benchmark results might not be too reliable since I might have made some mistakes. I added updated benchmark results [kornia/kornia#702 (comment)](https://github.com/kornia/kornia/issues/702#issuecomment-1451323769). If I scale the numbers correctly, I think your CPU results are faster than mine, and my GPU results are faster than yours. The difference could be due to some mistakes in my timing script (you can check in the comment) or different hardware.

yep yep, my GPU is a 3060 and yours is a 3090, so in seconds, I expected this difference. But comparing the ratio I expected something closer -- but it's fine 😄

Reply:
I see. I didn't know that mid-end consumer GPU can have a different eager vs dynamo ratio. I can run your kornia benchmark script on 3090 if such a data point is important for you.

Reply:
it's the runner the problem, a silly mistake -- https://github.com/kornia/kornia-benchmark/pull/12 😁